Question: minimum version of maven is not defined - 3.0.5 or 3.3.9

The Simulator does not define a minimum version of maven, which might break the build process for you in the future.

Either just set it to a fixed version for now, where I'd recommend `3.3.9` due to your use of the plugin `org.jacoco:jacoco-maven-plugin`, which could be updated to version `0.8.6`.

Currently you state maven `3.0.5` as prerequisite in the readme.

If you'd decide on which version you'd want you'd help a lot in getting rid of those Error messages durring the build phase in recent maven versions.

I just tried using 3.0.5 it compiles and all tests work,
but the mvn:versions:[...] command does drop this:

> [ERROR] Project requires an incorrect minimum version of Maven.
[ERROR] Update the pom.xml to contain maven-enforcer-plugin to
[ERROR] force the Maven version which is needed to build this project.
[ERROR] See https://maven.apache.org/enforcer/enforcer-rules/requireMavenVersion.html
[ERROR] Using the minimum version of Maven: 3.0.5

But this might just be my system, as I'm using maven 3.6.3 as well.
I'll provide a PR for 3.0.5 for now but do still recommend the usage of 3.3.9.

--- a/pom.xml
+++ b/pom.xml
@@ -167,6 +167,26 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <version>3.0.0-M3</version>
+                <executions>
+                    <execution>
+                        <id>enforce-maven</id>
+                        <goals>
+                            <goal>enforce</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <requireMavenVersion>
+                                    <version>3.0.5</version>
+                                </requireMavenVersion>
+                            </rules>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
             <!-- disable the jar plugin because we need the assembly plugin -->
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>

